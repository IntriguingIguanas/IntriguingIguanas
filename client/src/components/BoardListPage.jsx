import React from 'react'
import { hashHistory } from 'react-router'
import { connect } from 'react-redux'
import BoardPage from './BoardPage.jsx'
import { createBoard } from '../actions/createBoard'
import { fetchBoards } from '../actions/fetchBoards'

export class BoardListPage extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      boardName: ''
    }

    this.handleBoardNameChange = this.handleBoardNameChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleBoardNameChange(e) {
    this.setState({
      boardName: e.target.value
    })
  }

  handleSubmit(e) {
    e.preventDefault()
    this.props.submitBoard(this.state.boardName)
  }

  renderBoard(board_id) {
    hashHistory.push('/lobby/' + board_id)
  }

  render() {
    console.log(this.props.boardlist, "PROPS");
    return ( 
      <div>
        <input 
          value={ this.state.boardName } 
          onChange={ this.handleBoardNameChange }
        />
        <button onClick={ this.handleSubmit } >Create Board</button>
        { this.props.boardlist.map((board) => (
          <div 
            onClick={ () => this.renderBoard(board.board_id) }
          > { board.boardName }</div>
        )) }
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    boardlist: state.fetchBoards.boardlist
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    submitBoard: (boardName) => { dispatch(createBoard(boardName)) },
    getBoards: () => { dispatch(fetchBoards()) }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(BoardListPage)