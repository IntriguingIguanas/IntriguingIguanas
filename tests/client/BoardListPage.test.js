import React from 'react'
import ReactDOM from 'react-dom'
import { App } from 'App'
import { shallow, mount } from 'enzyme'
import { BoardListPage } from 'BoardListPage'
import renderer from 'react-test-renderer'

describe('BoardListPage', () => {
  var boardList = [{boardName: "BOARD 1"}, {boardName: "BOARD 2"}]
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BoardListPage boardlist={ boardList }/>, div);
  });
  it('should render create board input', () => {
    const boardListPage = renderer.create(<BoardListPage boardlist={ boardList } />).toJSON();
    expect(boardListPage).toMatchSnapshot();
  })
})