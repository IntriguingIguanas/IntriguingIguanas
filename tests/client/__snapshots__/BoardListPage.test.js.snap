// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardListPage should render create board input 1`] = `
<div>
  <input
    onChange={[Function]}
    value=""
  />
  <button
    onClick={[Function]}
  >
    Create Board
  </button>
  <div
    onClick={[Function]}
  >
     
    BOARD 1
  </div>
  <div
    onClick={[Function]}
  >
     
    BOARD 2
  </div>
</div>
`;
